liveblog/liveblog
# [LBSD-2768] - Connect the new readers' polls UI from Default SEO theme with the backend

### Purpose

This PR aims to integrate the UI components for the readers interface, developed in [LBSD-2767](https://github.com/liveblog/liveblog/pull/1362), with the backend endpoints implemented in [LBSD-2765](https://github.com/liveblog/liveblog/pull/1361).The goal is to enable readers to vote on the polls attached to blogs.

### What has changed

- Wired the frontend UI components for polls with the backend endpoints to allow poll voting.
- Added mechanism to prevent readers from voting more than once using the approach from the [research in LBSD-2766](https://sofab.atlassian.net/browse/LBSD-2766).


### Steps to test

1. Navigate to the Editor's interface.
2. Click "Add content here" and select Poll. Fill the poll form and submit. The poll will appear on the timeline.
3. Click "Live" to see the Live Preview. You will see the poll UI when open.
4. Select an option for the poll and click vote. This will trigger an event to place the vote and increment the total votes count. The selection will also be saved in storage for persistence.


### Screenshots
#### When Poll is Open
![image](https://github.com/liveblog/liveblog/assets/14116447/823a9496-69ce-4fbd-89f5-73b7d49954cf)

#### When Reader Votes
![image](https://github.com/liveblog/liveblog/assets/14116447/afd7fead-2f01-4907-8047-1c41e5c8ebc7)
_Note: Since the selection persists, even when you refresh, you shouldn't be able to vote again and your selection will be highlighted for a visual cue._


Resolves: [LBSD-2768](https://sofab.atlassian.net/browse/LBSD-2768)

[LBSD-2767]: https://sofab.atlassian.net/browse/LBSD-2767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LBSD-2765]: https://sofab.atlassian.net/browse/LBSD-2765?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LBSD-2768]: https://sofab.atlassian.net/browse/LBSD-2768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
